test_utf16 fails on big-endian architectures

Found by running tests on Debian s390x:

```
/tmp/autopkgtest-lxc.k2eym0yr/downtmp/autopkgtest_tmp/tests/test_http_response.py:174: in _assert_response_values
    self.assertEqual(response.body, body_bytes)
E   AssertionError: b'\xff\xfeh\x00i\x00' != b'\xfe\xff\x00h\x00i'
```

--- a/tests/test_http_response_text.py
+++ b/tests/test_http_response_text.py
@@ -156,12 +156,13 @@ class TestTextResponse(TestResponse):
 
     def test_utf16(self):
         """Test utf-16 because UnicodeDammit is known to have problems with"""
+        body = b"\xff\xfeh\x00i\x00"
         r = self.response_class(
             "http://www.example.com",
-            body=b"\xff\xfeh\x00i\x00",
+            body=body,
             encoding="utf-16",
         )
-        self._assert_response_values(r, "utf-16", "hi")
+        self._assert_response_values(r, "utf-16", body)
 
     def test_invalid_utf8_encoded_body_with_valid_utf8_BOM(self):
         r6 = self.response_class(

